HLTrigger/special/plugins/HLTPixelTrackFilter.cc

```diff
@@ -60,12 +63,22 @@ void HLTPixelTrackFilter::fillDescriptions(edm::ConfigurationDescriptions& descr
 // member functions
 //
 // ------------ method called to produce the data  ------------
-bool HLTPixelTrackFilter::filter(edm::Event& event, const edm::EventSetup& iSetup) {
+bool HLTPixelTrackFilter::hltFilter(edm::Event& event,
+                                    const edm::EventSetup& iSetup,
+                                    trigger::TriggerFilterObjectWithRefs& filterproduct) const {
+  // All HLT filters must create and fill an HLT filter object,
+  // recording any reconstructed physics objects satisfying (or not)
```

Comment:
I think often times the "trigger objects" are added to the event only if the filter accepts the event. In this case they are added irrespective of the decision of the filter. Is that intended ?

Reply:
This is true often, but not always. It is up to the use case of the specific filter.